parsons/redash/redash.py

```diff
@@ -0,0 +1,168 @@
+import json
+import logging
+import requests
+import time
+
+from parsons.etl.table import Table
+
+logger = logging.getLogger(__name__)
+
+
+class Redash(object):
+    """
+    Instantiate Redash Class
+
+    `Args:`
+        base_url: str
+            The base url for your redash instance (excluding the final /)
+        user_api_key: str
+            The user API key found in the User's profile screen
+        pause_time int
+            Specify time between polling for refreshed queries (Defaults to 3 seconds)
+        verify: bool
+            For https requests, should the certificate be verified (Defaults to True)
+    `Returns:`
+        Redash Class
+    """
+    args = {'base_url': None,
+            'user_api_key': None,
+            'pause_time': 3,
+            'verify': True,
+            'query_id': None,
+            'params': None}
+
+    def __init__(self,
+                 base_url="http://localhost",
+                 user_api_key=None,
+                 pause_time=3,
+                 verify=True):
+        self.base_url = base_url
+        self.user_api_key = user_api_key
+        self.pause = pause_time   # seconds to delay
+        self.verify = verify  # for https requests
+        self.session = requests.Session()
+        if user_api_key:
+            self.session.headers.update({'Authorization': f'Key {user_api_key}'})
+
+    def _poll_job(self, session, job, query_id):
+        while job['status'] not in (3, 4):
```

Comment:
It might be good to add a timeout arg to the class to allow users to specify the max time they're willing to wait.. and exit the loop accordingly.